src/Magnum/Shaders/FlatGL.h

```diff
@@ -1038,6 +1052,22 @@ template<UnsignedInt dimensions> class MAGNUM_SHADERS_EXPORT FlatGL: public GL::
         #endif
 };
 
+template<UnsignedInt dimensions> class FlatGL<dimensions>::CompileState : public FlatGL<dimensions> {
+public:
+    using FlatGL::isLinkFinished;
```

Comment:
Is this `using` even needed?

Reply:
I removed `using FlatGL::FlatGL` because we now construct the `FlatGL` instance separately and call `attachShaders`, `submitLink`, etc directly on it. The only public function that is really needed, is `isLinkFinished`, I guess.

Reply:
Only if we extend `FlatGL` by `private`/`protected`. This would allow us to hide all redundant FlatGL/ASP functions, as `isLinkFinished` is the only function needed in the CompileState, I guess. Is this a viable idea?

Reply:
It would be useful to have access to `flags()`, `materialCount()` etc., even if just for testing/diagnostic purposes (for example, if a certain async compilation takes *really long*, you could use these to discover it's due to the driver taking a lot of time to unroll a large loop).

Which means, you'd have to add `using FlatGL::flags;` etc., and then I feel the maintenance overhead is not worth it. Just keep everything public. If it ever becomes a footgun, we can reconsider, but right now I don't see a problem with having everything public :)